### PR TITLE
Mongo Shell Fixes

### PR DESCRIPTION
**Pull request by srinathnarayanan**
This PR
* Adds "isVirtualNetworkFilterEnabled" and "ipRules" to DatabaseAccount properties
* Replaces older mongo shell with new Mongo Terminal Shell for Mongo accounts
  - When notebooks are enabled
  - When VNET is not present
* Hides "Open Mongo Shell" when VNET is present
* Allows opening multiple terminal tabs (for all types of terminals)
* Polished Cassandra, Gremlin, Mongo command bars

mongoShell

[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/738)

### CODE REVIEW

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx [hunk 1]
```diff
@@ -134,13 +134,19 @@ export function createContextCommandBarButtons(container: Explorer): CommandButt
   const buttons: CommandButtonComponentProps[] = [];
 
   if (!container.isDatabaseNodeOrNoneSelected() && userContext.apiType === "Mongo") {
-    const label = "New Shell";
+    const showMongoTerminal =
+      container.isNotebookEnabled() && !userContext.databaseAccount.properties.isVirtualNetworkFilterEnabled;
```
Comment (swviswan) [inline src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:138]: >isVirtualNetworkFilterEnabled [](http://example.com/codeflow?start=79&length=29)

may be we need addnal property check for iprules

**Review by gitanuj — APPROVED** [review]

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx (hunk 1 above)
Comment (srinathnarayanan) [reply src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:138]: I checked tha iprules array is populated only when isVirtualNetworkFilterEnabled is true. Any particular reason to add ipRUles check as well?

Comment (gitanuj) [reply src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:138]: Is there a possibility that the account has VNET disabled but has firewall enabled? If yes we may need additional check here.

File: src/Explorer/Explorer.tsx [hunk 2]
```diff
@@ -1848,32 +1848,27 @@ export default class Explorer {
         throw new Error("Terminal kind: ${kind} not supported");
     }
 
-    const terminalTabs: TerminalTab[] = this.tabsManager.getTabs(
-      ViewModels.CollectionTabKind.Terminal,
-      (tab) => tab.hashLocation() == hashLocation
+    const terminalTabs: TerminalTab[] = this.tabsManager.getTabs(ViewModels.CollectionTabKind.Terminal, (tab) =>
+      tab.hashLocation().includes(hashLocation)
```
Comment (swviswan) [inline src/Explorer/Explorer.tsx:1852]: >includes [](http://example.com/codeflow?start=25&length=8)

startswith

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx [hunk 3]
```diff
@@ -70,7 +70,11 @@ export function createStaticCommandBarButtons(container: Explorer): CommandButto
       buttons.push(createEnableNotebooksButton(container));
     }
 
-    if (userContext.apiType === "Mongo") {
+    if (
+      userContext.apiType === "Mongo" &&
+      !userContext.databaseAccount.properties.isVirtualNetworkFilterEnabled &&
```
Comment (swviswan) [inline src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:75]: >isVirtualNetworkFilterEnabled [](http://example.com/codeflow?start=46&length=29)

use common method .. need ot also check for notebooks enabled

File: src/Explorer/Explorer.tsx [hunk 4]
```diff
@@ -246,9 +249,15 @@ export default class Explorer {
           async () => {
             this.isNotebookEnabled(
               userContext.authType !== AuthType.ResourceToken &&
-                ((await this._containsDefaultNotebookWorkspace(userContext.databaseAccount)) ||
-                  userContext.features.enableNotebooks)
+              ((await this._containsDefaultNotebookWorkspace(userContext.databaseAccount)) ||
+                userContext.features.enableNotebooks)
+            );
+            this.isTerminalEnabled(
```
Comment (swviswan) [inline src/Explorer/Explorer.tsx:255]: >isTerminalEnabled [](http://example.com/codeflow?start=17&length=17)

isShellEnabled?

Comment (swviswan) [inline src/Explorer/Explorer.tsx:255]: >isTerminalEnabled [](http://example.com/codeflow?start=17&length=17)

add isRunningOnNationalCloud check?

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx [hunk 5]
```diff
@@ -61,48 +61,36 @@ export function createStaticCommandBarButtons(container: Explorer): CommandButto
...
+      buttons.push(createEnableNotebooksButton(container));
+    }
   }
 
   if (!container.isDatabaseNodeOrNoneSelected()) {
-    if (container.isNotebookEnabled()) {
-      buttons.push(createDivider());
-    }
-
     const isSqlQuerySupported = userContext.apiType === "SQL" || userContext.apiType === "Gremlin";
```
Comment (swviswan) [inline src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:82]: >isSqlQuerySupported [](http://example.com/codeflow?start=10&length=19)

isOpenQuerySupported?

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx [hunk 6]
```diff
@@ -61,48 +61,36 @@ export function createStaticCommandBarButtons(container: Explorer): CommandButto
...
-  if (!container.isRunningOnNationalCloud()) {
-    if (!container.isNotebookEnabled()) {
-      buttons.push(createEnableNotebooksButton(container));
-    }
+    buttons.push(createOpenTerminalButton(container));
 
-    if (userContext.apiType === "Mongo") {
+    buttons.push(createNotebookWorkspaceResetButton(container));
+    if (userContext.apiType === "Mongo" && container.isTerminalEnabled() && container.isDatabaseNodeOrNoneSelected()) {
       buttons.push(createOpenMongoTerminalButton(container));
```
Comment (swviswan) [inline src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:69]: >createOpenMongoTerminalButton [](http://example.com/codeflow?start=19&length=29)

as discussed, need spacer between notebooks related commands and shell

**Review by swviswan — APPROVED** [review]

**Pull request merged**